src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -246,15 +246,15 @@ private static TextDocument CreateAdditionalDocument(DocumentId documentId, Proj
         /// </summary>
         public bool TryGetCompilation(out Compilation compilation)
         {
-            return _solution.TryGetCompilation(this.Id, out compilation);
+            return _solution.State.TryGetCompilation(this.Id, out compilation);
```

Comment:
you mean access thorough state? solution/project/document already access its state each other.